### PR TITLE
Optimise bioentities and test executable in mac M1

### PR DESCRIPTION
**Pull request by pcm32**
Adds logic for optimising bioentities collection after filling it.

I also ended up adding support for testing on M1 macs:
- Add docker platform line to run some containers as linux/amd64 when the container is only available for x86.
- Use tags that are available both for x86/amd64 and arm64 (M1), then it pulls the correct one for the arch.

### CODE REVIEW

**Review by alfonsomunozpomer — COMMENTED** [review]
There is probably a better test that we could run, which actually checks that the collection is otpimised. The steps would be:
1. Build the URL of each collection shard from the information returned by [`CLUSTERSTATUS`](https://solr.apache.org/guide/7_1/collections-api.html#clusterstatus) and some `jq` magic:
```bash
curl "http://${HOST}/solr/admin/collections?action=CLUSTERSTATUS&collection=${COLLECTION}" | jq '..|.replicas? | select( . != null ) | to_entries | .[] | .value | (.base_url|tostring)+"/admin/cores?action=STATUS&core="+(.core|tostring)'
```
This is necessary because optimisation status is stored in each core/shard/replica, not at the collection level. We need to build the URL by combining the core with the shards/replicas it stores. If we query a core for a shard/replica it doesn’t store we get an empty response.

2. Query each of the cores to get its optimisation status (i.e. `deletedDocs`):
```bash
xargs curl -s | jq '..|.deletedDocs? | select( . != null )'
```

3. And check that the value is only 0:
```bash
[ "$result" -eq 0 ]
```

So I think this could look like:
```bats
@test "[bioentities] Optimise collection" {
  if [ -z ${SOLR_HOST+x} ]; then
    skip "SOLR_HOST not defined, skipping load to Solr"
  fi

  run optimise-bioentities.sh

  # result stores the number of deletedDocs for each core/shard of the collection
  result=$(curl "http://${HOST}/solr/admin/collections?action=CLUSTERSTATUS&collection=${COLLECTION}" | jq '..|.replicas? | select( . != null ) | to_entries | .[] | .value | (.base_url|tostring)+"/admin/cores?action=STATUS&core="+(.core|tostring)' | xargs curl -s | jq '..|.deletedDocs? | select( . != null )' | uniq)

  echo "output = ${output}"
  [ "${result}" -eq 0 ]
}
```

**Comment by pcm32** [comment]
I think that the second curl is not getting an URL...

File: tests/bioentities.bats [hunk 1]
```diff
@@ -209,6 +209,19 @@ setup() {
...
 }
 
+@test "[bioentities] Check that optimisation worked" {
+  if [ -z ${SOLR_HOST+x} ]; then
+    skip "SOLR_HOST not defined, skipping load to Solr"
+  fi
+  COLLECTION=bioentities-v1
+
+  # result stores the number of deletedDocs for each core/shard of the collection
+  run curl "http://${SOLR_HOST}/solr/admin/collections?action=CLUSTERSTATUS&collection=${COLLECTION}" | jq '..|.replicas? | select( . != null ) | to_entries | .[] | .value | (.base_url|tostring)+"/admin/cores?action=STATUS&core="+(.core|tostring)' | xargs curl -s | jq '..|.deletedDocs? | select( . != null )' | uniq
```
Comment (alfonsomunozpomer) [inline tests/bioentities.bats]: Shouldn’t this be `result=$(cmd)`?

Comment (pcm32) [reply tests/bioentities.bats]: I think that bats needs at least one run statement. I noticed that apparently the run statement didn't like the pipings. Putting everything on a separate bash script and then run of that works.

**Review by alfonsomunozpomer — APPROVED** [review]
All good as far as I can see.

**Pull request merged**